File: app/src/main/java/com/pondipb/pondibrian/ukulimaiq/AgronomyActivity.java
package com.pondipb.pondibrian.ukulimaiq;

import android.os.Bundle;
import android.support.design.widget.FloatingActionButton;
import android.support.design.widget.Snackbar;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.support.v7.widget.Toolbar;
import android.view.View;

import java.util.Vector;

public class AgronomyActivity extends AppCompatActivity {
    //Recycler view field
    RecyclerView recyclerView;

    //Vector for videos URL
    Vector<YoutubeVideo>youtubeVideos = new Vector<YoutubeVideo>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_agronomy);

        recyclerView = findViewById(R.id.recyclerView);
        recyclerView.setHasFixedSize(true);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));

        //Load video list
        youtubeVideos.add(new YoutubeVideo());
        youtubeVideos.add(new YoutubeVideo());
        youtubeVideos.add(new YoutubeVideo());
        youtubeVideos.add(new YoutubeVideo());
        youtubeVideos.add(new YoutubeVideo());

        VideoAdapter videoAdapter = new VideoAdapter(youtubeVideos);

        recyclerView.setAdapter(videoAdapter);
    }

}

File: app/src/main/kotlin/com/pondipb/pondibrian/ukulimaiq/AgronomyActivity.kt
package com.pondipb.pondibrian.ukulimaiq

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

class AgronomyActivity : AppCompatActivity() {
    private lateinit var recyclerView: RecyclerView
    private val youtubeVideos = mutableListOf<YoutubeVideo>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_agronomy)

        recyclerView = findViewById(R.id.recyclerView)
        recyclerView.setHasFixedSize(true)
        recyclerView.layoutManager = LinearLayoutManager(this)

        // Load video list with improved embedded YouTube videos
        loadVideos()

        val videoAdapter = VideoAdapter(youtubeVideos)
        recyclerView.adapter = videoAdapter
    }

    private fun loadVideos() {
        // Agricultural best practices videos
        youtubeVideos.add(
            YoutubeVideo(
                videoUrl = getEmbedVideoHtml("dQw4w9WgXcQ"),
                title = "Sustainable Farming Practices",
                description = "Learn about modern sustainable farming techniques"
            )
        )
        youtubeVideos.add(
            YoutubeVideo(
                videoUrl = getEmbedVideoHtml("dQw4w9WgXcQ"),
                title = "Crop Rotation Methods",
                description = "Understanding effective crop rotation"
            )
        )
        youtubeVideos.add(
            YoutubeVideo(
                videoUrl = getEmbedVideoHtml("dQw4w9WgXcQ"),
                title = "Pest Management",
                description = "Integrated pest management strategies"
            )
        )
        youtubeVideos.add(
            YoutubeVideo(
                videoUrl = getEmbedVideoHtml("dQw4w9WgXcQ"),
                title = "Soil Health",
                description = "Maintaining and improving soil quality"
            )
        )
        youtubeVideos.add(
            YoutubeVideo(
                videoUrl = getEmbedVideoHtml("dQw4w9WgXcQ"),
                title = "Water Management",
                description = "Efficient irrigation and water conservation"
            )
        )
    }

    private fun getEmbedVideoHtml(videoId: String): String {
        return """
            <iframe width="100%" height="100%" 
                src="https://www.youtube.com/embed/$videoId" 
                frameborder="0" 
                allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" 
                allowfullscreen>
            </iframe>
        """.trimIndent()
    }
}
